manual/retries/README.md

```diff
@@ -1,6 +1,201 @@
+of replicas that where *required* by the requested consistency level.
+
+#### [onReadTimeout]
+
+A read request reached the coordinator, which initially believed that there were enough live replicas to process it.
+But, for some reason, one or several replicas were too slow to answer within the predefined timeout
+(`read_request_timeout_in_ms` in `cassandra.yaml`), and the coordinator replied to the client with a `READ_TIMEOUT`
+error.
+
+This could be due to a temporary overloading of these replicas, or even
+that they just failed or were turned off. During reads, Cassandra doesn't request data from every replica to minimize
+internal network traffic; instead, some replicas are only asked for a checksum of the data. A read timeout may occur
+even if enough replica responded to fulfill the consistency level, but only checksum responses were received (the
+method's `dataRetrieved` parameter allow you to check if you're in that situation).
+
+If the policy rethrows the error, the user code will get a [ReadTimeoutException].
+
+Note: do not confuse this error with a [driver read timeout], which happens when the coordinator didn't reply at all to
+the client.
+
+#### [onWriteTimeout]
+
+This is similar to `onReadTimeout`, but for write operations. The reason reads and writes are handled separately is
+because a read is obviously a non mutating operation, whereas a write is likely to be. If a write times out at the
+coordinator level, there is no way to know whether the mutation was applied or not on the non-answering replica.
+Policy implementations are usually more conservative in `onWriteTimeout` in case the write was not idempotent.
+
+If the policy rethrows the error, the user code will get a [WriteTimeoutException].
+
+#### [onRequestError]
+
+This gets called for any other error occurring after the request was sent.
+
+Note: this method is specified by [ExtendedRetryPolicy], a child interface of `RetryPolicy`. This is for backward
+compatibility: `onRequestError` did not exist in previous 2.1 versions, so it wasn't added directly on `RetryPolicy` to
+avoid breaking legacy implementations. All built-in policies implement `ExtendedRetryPolicy`, and we recommend that you
+do too. If you pass a policy that only implements `RetryPolicy`, the driver will use a default `onRequestError`
+implementation that always retries on the next host with the same consistency level (this is consistent with what
+previous 2.1 versions did). In 3.0+, the two interfaces have been merged back together.
+
+The method receives the exception as a parameter, so that implementations can refine their decision based on what
+happened. The possible exceptions are:
+
+* [ServerError]: thrown by the coordinator when an unexpected error occurs. This is generally a Cassandra bug;
+* [OperationTimedOutException]: thrown by the client when it didn't hear back from the coordinator within the
```

Comment:
So are we really planning to turn `OperationTimedOutException` and `ConnectionException` public?